src/ODETools/ODEOperators.jl

```diff
@@ -8,76 +8,88 @@ struct Nonlinear <: OperatorType end
 """
 const AffineODEOperator = ODEOperator{Affine}
-# abstract type AffineODEOperator <: ODEOperator end
 
 """
 It represents a constant operator in an implicit ODE, i.e., an ODE operator of
-the form A(t,u,∂tu) = M∂tu + Ku + f
+the form A(t,u,∂tu,...,,∂t^Nu) = A_N∂t^Nu + ...A_1∂tu + A_0u + f
 """
 const ConstantODEOperator = ODEOperator{Constant}
 
-# @santiagobadia : I would consider in a future a more general case, in which
-# the implicit ODE has an arbitrary order, i.e., A(t,u,u_t, u_nt) = 0.
-# get_order(op::ODEOperator) = @notimplemented
-# @santiagobadia :
-# We probably want to consider second order time derivatives too, i.e.,
-# A(t,u,u_t,u_tt) = 0 (wave propagation, elastodynamics) or even more...
 """
 Returns the `OperatorType`, i.e., nonlinear, affine, or constant in time
 """
-OperatorType(::ODEOperator{C}) where C = C
-# OperatorType(::Type{<:ODEOperator{C}}) where C = C
+OperatorType(::ODEOperator{C}) where {C} = C
 
 """
-It provides A(t,u,∂tu) for a given (t,u,∂tu)
+Returns the order of the ODE operator
 """
-function residual!(r::AbstractVector,op::ODEOperator,t::Real,u::AbstractVector,u_t::AbstractVector,ode_cache)
+function get_order(::ODEOperator)
   @abstractmethod
 end
 
 """
+It provides A(t,u,∂tu,...,,∂t^Nu) for a given (t,u,∂tu,...,,∂t^Nu)
 """
-function allocate_residual(op::ODEOperator,u::AbstractVector,ode_cache)
+function residual!(
+  r::AbstractVector,
+  op::ODEOperator,
+  t::Real,
+  u::Union{AbstractVector,Tuple{Vararg{AbstractVector}}},
+  ode_cache)
   @abstractmethod
 end
 
 """
-It adds [∂A/∂u](t,u,∂tu) for a given (t,u,∂tu) to a given matrix J
 """
-function jacobian!(J::AbstractMatrix,op::ODEOperator,t::Real,u::AbstractVector,u_t::AbstractVector,ode_cache)
+function allocate_residual(
+  op::ODEOperator,
+  u::Union{AbstractVector,Tuple{Vararg{AbstractVector}}},
+  ode_cache)
   @abstractmethod
-  # Add values to J
 end
 
 """
-Add the contribution γ*[∂A/∂(∂tu)](u,∂tu) to the Jacobian matrix, where γ
-is a scaling coefficient provided by the `ODESolver`, e.g., 1/Δt for Backward
-Euler; It represents ∂(δt(u))/∂(u), in which δt(⋅) is the approximation of ∂t(⋅)
-in the solver.
-"""
-function jacobian_t!(J::AbstractMatrix,op::ODEOperator,t::Real,u::AbstractVector,u_t::AbstractVector,dut_u::Real,ode_cache)
+It adds γ_i*[∂A/∂(∂t^iu)](t,u,∂tu,...,,∂t^Nu) for a given (t,u,∂tu,...,∂t^Nu) to
+a given matrix J, where γ is a scaling coefficient provided by the `ODESolver`,
+e.g., 1/Δt for Backward Euler; It represents ∂(δt^i(u))/∂(u), in which δt^i(⋅) is
+the approximation of ∂t^i(⋅) in the solver.
+"""
+function jacobian!(
+  J::AbstractMatrix,
+  op::ODEOperator,
+  t::Real,
+  u::Tuple{Vararg{AbstractVector}},
+  i::Integer,
+  γᵢ::Real,
+  ode_cache)
   @abstractmethod
   # Add values to J
 end
 
 """
-Add the contribution of both `jacobian!` and `jacobian_t!`,i.e.,
-[∂A/∂u](t,u,∂tu)+γ*[∂A/∂(∂tu)](u,∂tu)
+Add the contribution of both all jacobians ,i.e., ∑ᵢ γ_i*[∂A/∂(∂t^iu)](t,u,∂tu,...,,∂t^Nu)
```

Comment:
both out

Reply:
done in https://github.com/gridap/GridapODEs.jl/commit/8e371a3e8be38ed1b2875c75e873e2c0db7c62de